src/lib/filesystem/local.ts

```diff
@@ -0,0 +1,59 @@
+import * as webnative from 'webnative'
+import { createEmitter, type FileSystem as FileSystemEvents } from 'webnative/events'
+import { isCID } from 'webnative/fs/types/check'
+import { type CID, decodeCID, encodeCID } from 'webnative/common/cid'
+import { FileSystem, type Program } from 'webnative'
+
+const LOCAL_ONLY_FS_KEY = 'local-only-fs'
+
+function setupHooks(fs: webnative.FileSystem, program: Program): void {
+  fs.publish = async function () {
+    const cid: CID = await this.root.put()
+    await program.components.storage.setItem(LOCAL_ONLY_FS_KEY, encodeCID(cid))
+    return cid
+  }
+}
+
+async function localOnlyFileSystem(program: Program): Promise<webnative.FileSystem> {
```

Comment:
nice 👌🏼 only thing i'd suggest is maybe adding a comment to make it more clear that this function will first check for an `existingFs` and only return a `newFs` if an existing one isn't found

Reply:
Yeah good call! 💯 Added a comment.